spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -1,412 +0,0 @@
-    spark
-        .read()
-        .schema(sparkSchema)
-        .json(JavaSparkContext.fromSparkContext(spark.sparkContext()).parallelize(data0))
-        .write()
-        .format("iceberg")
-        .mode(SaveMode.Append)
-        .save(targetPath);
-
-    // read from parquet and append to iceberg w/ nullability check disabled
-    spark
-        .read()
-        .schema(SparkSchemaUtil.convert(icebergSchema))
-        .parquet(sourcePath)
-        .write()
-        .format("iceberg")
-        .option(SparkWriteOptions.CHECK_NULLABILITY, false)
-        .mode(SaveMode.Append)
-        .save(targetPath);
-
-    // read all data
-    List<Row> rows = spark.read().format("iceberg").load(targetPath).collectAsList();
-    assumeThat(rows).as("Should contain 6 rows").hasSize(6);
-  }
-
-  @TestTemplate
-  public void testNullableWithSparkSqlOption() throws IOException {
-    assumeThat(spark.version())
-        .as("Spark 3 rejects writing nulls to a required column")
-        .startsWith("2");
-
-    String sourcePath = String.format("%s/nullable_poc/sourceFolder/", location);
-    String targetPath = String.format("%s/nullable_poc/targetFolder/", location);
-
-    tableProperties = ImmutableMap.of(TableProperties.WRITE_DATA_LOCATION, targetPath);
-
-    // read this and append to iceberg dataset
-    spark
-        .read()
-        .schema(sparkSchema)
-        .json(JavaSparkContext.fromSparkContext(spark.sparkContext()).parallelize(data1))
-        .write()
-        .parquet(sourcePath);
-
-    SparkSession newSparkSession =
-        SparkSession.builder()
-            .master("local[2]")
-            .appName("NullableTest")
-            .config(SparkSQLProperties.CHECK_NULLABILITY, false)
-            .getOrCreate();
-
-    // this is our iceberg dataset to which we will append data
-    new HadoopTables(newSparkSession.sessionState().newHadoopConf())
-        .create(
-            icebergSchema,
-            PartitionSpec.builderFor(icebergSchema).identity("requiredField").build(),
-            tableProperties,
-            targetPath);
-
-    // this is the initial data inside the iceberg dataset
-    newSparkSession
-        .read()
-        .schema(sparkSchema)
-        .json(JavaSparkContext.fromSparkContext(spark.sparkContext()).parallelize(data0))
-        .write()
-        .format("iceberg")
-        .mode(SaveMode.Append)
-        .save(targetPath);
-
-    // read from parquet and append to iceberg
-    newSparkSession
-        .read()
-        .schema(SparkSchemaUtil.convert(icebergSchema))
-        .parquet(sourcePath)
-        .write()
-        .format("iceberg")
-        .mode(SaveMode.Append)
-        .save(targetPath);
-
-    // read all data
-    List<Row> rows = newSparkSession.read().format("iceberg").load(targetPath).collectAsList();
-    assumeThat(rows).as("Should contain 6 rows").hasSize(6);
-  }
-
-  @TestTemplate
-  public void testFaultToleranceOnWrite() throws IOException {
```

Comment:
I dropped this test because it is testing basic Spark behavior and doesn't belong in scan and write tests for specific schemas. I didn't move it anywhere because I don't think it is a valuable test. Spark stage failures throw exceptions and don't commit. I think it was originally trying to check for side-effects, but that isn't necessary in Iceberg.